### PR TITLE
Use 1px buffers

### PR DESCRIPTION
**Pull request by Leon-Plickat**
untested for now.

### CODE REVIEW

**Comment by Leon-Plickat** [comment]
On Fri Apr 29, 2022 at 1:29 PM CEST, Isaac Freund wrote:
> Can we create a single buffer during initialization that we share
> between all outputs to avoid this failure mode?

I am not sure attaching a single buffer to multiple surfaces is
allowed, I'll have to read the core protocol again.

**Comment by ifreund** [comment]
> I am not sure attaching a single buffer to multiple surfaces is allowed, I'll have to read the core protocol again.

The only limitation there is that if a buffer is attached to multiple surfaces wl_buffer.release events become unreliable/meaningless. It's suggested to create multiple buffers with the same backing storage if one cares about the release events, we don't though.

File: src/Lock.zig [hunk 1]
```diff
@@ -382,3 +398,36 @@ fn fatal_oom() noreturn {
...
+    }
+
+    const pool = try lock.shm.?.createPool(fd, @intCast(i32, size));
+    defer pool.destroy();
+
+    return try pool.createBuffer(
+        0,
+        @intCast(i32, width),
+        @intCast(i32, height),
+        @intCast(i32, stride),
```
Comment (ifreund) [inline src/Lock.zig:430]: these int casts shouldn't be needed.

File: src/Output.zig [hunk 2]
```diff
@@ -31,10 +34,14 @@ pub fn create_surface(output: *Output) !void {
     const lock_surface = try output.lock.session_lock.?.getLockSurface(surface, output.wl_output);
     lock_surface.setListener(*Output, lock_surface_listener, output);
     output.lock_surface = lock_surface;
+
+    const viewport = try output.lock.viewporter.?.getViewport(surface);
+    output.viewport = viewport;
```
Comment (ifreund) [inline src/Output.zig:39]: unlike the lines above, line length isn't an issue here.
```suggestion
    output.viewport = try output.lock.viewporter.?.getViewport(surface);
```

**Review by ifreund — APPROVED** [review]
LGTM, thanks! Just cleaned up the code and commit message a bit before merging

**Comment by Leon-Plickat** [comment]
Cool, I'll change it to a single buffer then.

**Pull request merged**